snapcraft/internal/repo.py

```diff
@@ -546,3 +549,21 @@ def _try_copy_local(path, target):
 def check_for_command(command):
     if not shutil.which(command):
         raise MissingCommandError([command])
+
+
+def _get_pkg_name_parts(pkg_name):
+    """Break package name into base parts"""
+
+    name = pkg_name
+    version = None
+    if '=' in pkg_name:
+        name, version = pkg_name.split('=')
```

Comment:
Small nit: this isn't very pythonic, and checks for '=' twice. Something like this might be better on both fronts:

```python
def _get_pkg_name_parts(pkg_name):
    name = pkg_name
    version = None
    with contextlib.suppress(ValueError):
        name, version = pkg_name.split('=')
    return name, version
```